delta-app/src/main/java/io/cdap/delta/store/DBReplicationStateStore.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.store;
+
+import io.cdap.cdap.api.dataset.lib.CloseableIterator;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.TableNotFoundException;
+import io.cdap.cdap.spi.data.table.StructuredTableId;
+import io.cdap.cdap.spi.data.table.StructuredTableSpecification;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.FieldType;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.table.field.Range;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A StructuredTable based database table to store generic replicator states.
+ */
+public class DBReplicationStateStore {
+
+  private static final StructuredTableId TABLE_ID = new StructuredTableId("delta_state_store");
+  private static final String NAMESPACE_COL = "namespace";
+  private static final String GENERATION_COL = "generation";
+  private static final String APPNAME_COL = "app_name";
+  private static final String INSTANCEID_COL = "instance_id";
+  private static final String STATE_COL = "state";
+  private static final String VALUE_COL = "value";
+  private static final String UPDATED_COL = "last_updated";
+
+  public static final StructuredTableSpecification TABLE_SPEC = new StructuredTableSpecification.Builder()
+    .withId(TABLE_ID)
+    .withFields(new FieldType(NAMESPACE_COL, FieldType.Type.STRING),
+                new FieldType(GENERATION_COL, FieldType.Type.LONG),
+                new FieldType(APPNAME_COL, FieldType.Type.STRING),
+                new FieldType(INSTANCEID_COL, FieldType.Type.INTEGER),
+                new FieldType(STATE_COL, FieldType.Type.STRING),
+                new FieldType(VALUE_COL, FieldType.Type.STRING),
+                new FieldType(UPDATED_COL, FieldType.Type.LONG))
+    .withPrimaryKeys(NAMESPACE_COL, APPNAME_COL, GENERATION_COL, INSTANCEID_COL, STATE_COL)
+    .build();
+
+  private final StructuredTable table;
+
+  private DBReplicationStateStore(StructuredTable table) {
+    this.table = table;
+  }
+
+  static DBReplicationStateStore get(StructuredTableContext context) {
```

Comment:
The intention was the keep this `get` method usable with the package only. 

While the class is public because of the `StructuredTableSpecification TABLE_SPEC` is being used by `AssessmentService` , which is outside of the package.